src/CineFiles.jl

```diff
@@ -2,19 +2,219 @@ module CineFiles
 
 using ColorTypes: N0f8, N4f12, Gray
 using LRUCache
+using StaticArrays
 
-import Base: eltype, length, size, getindex, firstindex, lastindex, iterate
+import Base: eltype, length, size, getindex, firstindex, lastindex, iterate, read
 export CineFile
 
 # using FileIO
 # add_format(format"CINE", [0x43, 0x49], ".cine")
 
-struct Time64
+abstract type BinaryData end
+
+read(f::IOStream, S::Type{T}) where {T<:BinaryData} = S(read.(Ref(f), S.types)...)
+read(f::AbstractString, S::Type{T}) where {T<:BinaryData} = S(read.(f, S.types)...)
+
+struct Time64 <: BinaryData
     fractions::UInt32
     seconds::UInt32
 end
 
-struct CineFileHeader
+struct WBgain <: BinaryData
+    R::Float32
+    B::Float32
+end
+
+struct IMfilter <: BinaryData
+    Dim::Int32
+    Shifts::Int32
+    Bias::Int32
+    Coef::SizedMatrix{5,5,Int32,2,Matrix{Int32}}
+end
+
+struct Rect <: BinaryData
+    Left::Int32
+    Top::Int32
+    Right::Int32
+    Bottom::Int32
+end
+
+struct TimeCode <: BinaryData
+    framesU::UInt8
+    framesT::UInt8
+    dropFrameFlag::UInt8
+    colorFrameFlag::UInt8
+    secondsU::UInt8
+    secondsT::UInt8
+    flag1::UInt8
+    minutesU::UInt8
+    minutesT::UInt8
+    flag2::UInt8
+    hoursU::UInt8
+    hoursT::UInt8
+    flag3::UInt8
+    flag4::UInt8
+    userBitData::UInt32
+end
+
+struct SetupHeader <: BinaryData
+    FrameRate16::UInt16
+    Shutter16::UInt16
+    PostTrigger16::UInt16
+    FrameDelay16::UInt16
+    AspectRatio::UInt16
+    Res7::UInt16
+    Res8::UInt16
+    Res9::UInt8
+    Res10::UInt8
+    Res11::UInt8
+    TrigFrame::UInt8
+    Res12::UInt8
+    DescriptionOld::SizedVector{121,Int8}
+    Mark::UInt16
+    Length::UInt16
+    Res13::UInt16
+    SigOption::UInt16
+    BinChannels::Int16
+    SamplesPerImage::UInt8
+    BinName::SizedMatrix{8,11,Int8}
+    AnaOption::UInt16
+    AnaChannels::Int16
+    Res6::UInt8
+    AnaBoard::UInt8
+    ChOption::SizedVector{8,Int16}
+    AnaGain::SizedVector{8,Float32}
+    AnaUnit::SizedMatrix{8,6,Int8}
+    AnaName::SizedMatrix{8,11,Int8}
+    lFirstImage::Int32
+    dwImageCount::UInt32
+    nQFactor::Int16
+    wCineFileType::UInt16
+    szCinePath::SizedMatrix{4,65,Int8}
+    Res14::UInt16
+    Res15::UInt8
+    Res16::UInt8
+    Res17::UInt16
+    Res18::Float64
+    Res19::Float64
+    Res20::UInt16
+    Res1::Int32
+    Res2::Int32
+    Res3::Int32
+    ImWidth::UInt16
+    ImHeight::UInt16
+    EDRShutter16::UInt16
+    Serial::UInt32
+    Saturation::Int32
+    Res5::UInt8
+    AutoExposure::UInt32
+    bFlipH::SizedVector{4,Bool}
```

Comment:
Bools are aliased to Int32s in the header:
```
typedef int bool32_t;
```
...but Julia `Bool`s are one byte apiece. Can you swap these out for Int32s?